File: p1431.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        // boolean[] result = new boolean[candies.length];
        List<Boolean> result = new ArrayList<>();

        int greatest = 0;
        for (int candy : candies){
            if (candy > greatest){
                greatest = candy;
            }
        }

        for (int i = 0; i < candies.length; i++){
            candies[i] += extraCandies;

            if (candies[i] >= greatest){
                result.add(i, true);
            } else{
                result.add(i, false);
            }
        }

        return result;
    }
}

File: p1431.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """

        greatest = 0

        for candy in candies:
            if candy > greatest:
                greatest = candy

        for index, candy in enumerate(candies):
            candy += extraCandies

            if candy >= greatest:
                candies[index] = True
            else:
                candies[index] = False
        
        return candies